src/Common/src/TypeSystem/Common/SignatureVariable.cs

```diff
@@ -64,7 +64,7 @@ public override int GetHashCode()
 
         protected override TypeFlags ComputeTypeFlags(TypeFlags mask)
         {
-            TypeFlags flags = 0;
+            TypeFlags flags = TypeFlags.ContainsGenericVariablesComputed | TypeFlags.ContainsGenericVariables;
```

Comment:
I think that's only because you have the `if` check that tries to prevent calling `InstantiateSignature` if there's nothing to instantiate - if you follow my suggestion, it shouldn't be needed.